### PR TITLE
Add constant yawspeed

### PR DESCRIPTION
**Pull request by khanghugo**

### CODE REVIEW

File: src/hlstrafe.cpp [hunk 1]
```diff
@@ -1998,6 +2011,9 @@ namespace HLStrafe
 		out.NextFrameIs0ms = false;
 
 		float yaw = static_cast<float>(NormalizeDeg(out.Yaw));
+
+		curState.LastYaw = yaw;
```
Comment (YaLTeR) [inline src/hlstrafe.cpp]: I think you forgot to commit hlstrafe.hpp changes

File: src/hlstrafe.cpp [hunk 2]
```diff
@@ -685,6 +685,12 @@ namespace HLStrafe
 			theta = right ? -theta : theta;
 
 			double yaw = vel_yaw - phi + theta;
+
+			if (curState.ConstantYawSpeed) {
+				auto max_yaw_delta = curState.ConstantYawSpeedValue * vars.Frametime;
+				yaw = (right ? curState.LastYaw - max_yaw_delta : curState.LastYaw + max_yaw_delta) * M_DEG2RAD;
```
Comment (YaLTeR) [inline src/hlstrafe.cpp]: so player.Viewangles[1] cannot be used here? It's already modified by something before this code?

File: src/hlstrafe.cpp (hunk 1 above)
Comment (YaLTeR) [reply src/hlstrafe.cpp]: Also it's a little weird to use a curState variable for something that seems to be fully local to MainFunc but it's understandable

File: src/hlstrafe.cpp (hunk 2 above)
Comment (khanghugo) [reply src/hlstrafe.cpp]: Hmmm I didn't think of that.

File: src/hlstrafe.cpp [hunk 3]
```diff
@@ -1998,6 +2027,7 @@ namespace HLStrafe
 		out.NextFrameIs0ms = false;
 
 		float yaw = static_cast<float>(NormalizeDeg(out.Yaw));
+
```
Comment (YaLTeR) [inline src/hlstrafe.cpp]: mhm

**Comment by YaLTeR** [comment]
Thanks

**Pull request merged**